diff --git a/arch/arm/dts/stm32mp15xx-dkx.dtsi b/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 58a068db39..27eff68db0 100644
--- a/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -144,9 +144,21 @@
 	status = "okay";
 };
 
+
+&pinctrl
+{
+	ethernet0_power_enable_pins: power-enable-0 {
+    	    power_enable {
+    	        pinmux = <STM32_PINMUX('E', 5, GPIO_ACTIVE_HIGH)>; /* PE5 as GPIO, High */
+    	        output-high; // This ensures the pin starts in a high state
+    	        bias-disable;
+    	    };
+	};
+};
+
 &ethernet0 {
 	status = "okay";
-	pinctrl-0 = <&ethernet0_rgmii_pins_a>;
+	pinctrl-0 = <&ethernet0_rgmii_pins_a> , <&ethernet0_power_enable_pins>;;
 	pinctrl-1 = <&ethernet0_rgmii_sleep_pins_a>;
 	pinctrl-names = "default", "sleep";
 	phy-mode = "rgmii-id";
